lib/src/rules/unnecessary_null_checks.dart

```diff
@@ -95,3 +75,33 @@ class _Visitor extends SimpleAstVisitor<void> {
     }
   }
 }
+
+DartType getExpectedType(PostfixExpression node) {
```

Comment:
Consider moving this to one of the utility libraries.

Reply:
I put this function here because it is really related to the lints (that is where _null check_ checks happen). If we move it in  utility library we should rename it to clearly indicate that it not only returns the expected type but it also only works where `null_check_on_nullable_type_parameter` and `unnecessary_null_checks` should apply.

Reply:
sgtm. It just feels like a code smell to have code being shared between lints that isn't in a utility library that's intended to be shared.

Reply:
To be clear, is it ok to let this function here?

Reply:
It doesn't impact the correctness of the lint, so I suppose so.

Reply:
So I think I addressed all your comments and this PR is ready for another look :)